### PR TITLE
[i18n] Added "Language and locale" to Settings

### PR DESCRIPTION
**Pull request by juan-g**
pull request #703

- added Language option to General Settings, just a text field for
flexibility, to allow any language
- click on the "Save settings" button, and restart Ghost to activate the
new language
- default: English (en); currently available: Spanish (es); translation
files can be added for any language
- the corresponding translation files should be at
_core/server/translations/en.json_, _es.json_... and
_content/themes/mytheme/assets/translations/mytheme_en.json_,
_mytheme_en.css_...
- `currentLocale`, dynamically based on overall settings (key =
"`default_lang`") in the `settings` db table
- during Ghost's initialization, settings available inside `i18n`
functions; see files _core/server/i18n.js_ and _index.js_
- there is a complementary commit for Ghost's PR TryGhost/Ghost#8437

### CODE REVIEW

File: app/models/setting.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default Model.extend(ValidationEngine, {
     logo: attr('string'),
     coverImage: attr('string'),
     icon: attr('string'),
-    defaultLang: attr('string'),
+    default_lang: attr('string'),
```
Comment (acburdine) [inline app/models/setting.js]: This should remain in camelCase format (e.g. what it was before). On the server side we use the underscore as a separator, but on the client side we use camelCase unless it's absolutely necessary.

Comment (juan-g) [inline app/models/setting.js]: @acburdine, OK, I'll look into it.

**Comment by juan-g** [comment]
[i18n] `defaultLang` on the client side

pull request #703

- using `defaultLang` on the client side, and keeping `default_lang` on the server side, as requested
- tested again, it works well

File: app/models/setting.js (hunk 1 above)
Comment (juan-g) [inline app/models/setting.js]: Done.

**Comment by juan-g** [comment]
Thank you very much for the approval. Just a little simplification in the help text because of easier working now:

[i18n] no need to restart Ghost after switching language

pull request #703

- now the change is immediate, there is no need to restart Ghost
- removed one of the two help lines (just this change here)
- there is a complementary commit for Ghost's PR TryGhost/Ghost#8437

**Comment by juan-g** [comment]
[i18n] Adapted to Ghost-Admin update: `defaultLang` -> `defaultLocale`

pull request #703

- also rebased
- there is a complementary commit for Ghost's PR TryGhost/Ghost#8437

**Comment by juan-g** [comment]
Commit after updating, rebasing, etc.:

[i18n] _"Language"_ setting renamed _"Language and locale"_

pull request #703

- added _"locale"_ to setting title and description
- removed mention to availability of Spanish translation files (Casper)

**Comment by juan-g** [comment]
Tests pass locally (it's just a minor text change), but here it says:

> The command "COVERAGE=true npm test" exited with 1.

**Comment by kirrg001** [comment]
@kevinansfield If you have a minute, could you please check why the tests fail on Travis?
@juan-g said, the tests passed locally. Thanks!

**Comment by kevinansfield** [comment]
@kirrg001, @juan-g not sure what was going on there but clearing the cache on Travis and restarting the build fixed it 😄

**Comment by juan-g** [comment]
Thank you! :+1:

**Comment by juan-g** [comment]
Updated local master from upstream, and rebased branch.

**Comment by kirrg001** [comment]
If i save any non existing locale in the admin panel, it's saved in the database, but on refresh, the input field is empty.

**Comment by juan-g** [comment]
Updated and rebased again.

**Comment by juan-g** [comment]
> If i save any non existing locale in the admin panel, it's saved in the database, but on refresh, the input field is empty.

It seems to work well now. Maybe it was fixed by updates.

**Comment by kevinansfield** [comment]
👍 functionally this looks fine, @kirrg001 I think this will be fine to merge as-is then we can edit the content once the help docs etc are online, we definitely want to link to those from the locale input's help text

**Comment by kirrg001** [comment]
Pre-link: https://themes.ghost.org/v1.20.0/docs/i18n
I need to edit the content tomorrow before the version get's marked as active.

**Comment by kirrg001** [comment]
@kevinansfield Would you like to add the link below the input field?

**Comment by kevinansfield** [comment]
rebased and pushed the updated copy

**Comment by kirrg001** [comment]
@kevinansfield Travis failed. I think it's not random.

**Comment by kevinansfield** [comment]
Just another random failure 😞

**Pull request merged**